autorest/adal/token.go

```diff
@@ -1024,13 +1027,18 @@ func (spt *ServicePrincipalToken) refreshInternal(ctx context.Context, resource
 
 // converts expires_on to the number of seconds
 func parseExpiresOn(s string) (json.Number, error) {
+	// convert the expiration date to the number of seconds from now
+	timeToDuration := func(t time.Time) json.Number {
+		dur := t.Sub(time.Now().UTC())
+		return json.Number(strconv.FormatInt(int64(dur.Round(time.Second).Seconds()), 10))
+	}
```

Comment:
@jhendrixMSFT, I'm a little confused why this calculates the difference between `t time.Time` (the passed in and parsed `s string`) and `time.Now().UTC()`. That difference I believe is effectively the `expires_in` value (the TTL of the token). 

However, when the `strconv.ParseInt(s, 10, 64)` logic is successful this `parseExpiresOn` function would be passing through the unix timestamp as far as I can tell (i.e. the actual time when the token expires and not the `expires_in` value).

From the IMDS docs - https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http

> `expires_on` - The timespan when the access token expires. The date is represented as the number of seconds from "1970-01-01T0:0:0Z UTC" (corresponds to the token's `exp` claim).

Reply:
I've moved this question to https://github.com/Azure/go-autorest/issues/638. Apologies for the noise here.